File: Kaighobadi_Neda/Aplicatie/Frontend/iTroc/src/app/product.ts

export class Product {
    constructor(
        public pic:string,
        public name:string,
        public category: string,
        public subcategory: string,
        public description:string,
        public price: number,
        public sellerId:string,
        public sellerAdress: string,
        public id?:string
    ) { }
    
    public editProduct(pic:string,name:string,category:string,subcategory:string,description:string,price:number){
        this.pic=pic;
        this.name=name;
        this.category=category;
        this.subcategory=subcategory;
        this.description=description;
        this.price=price;
    }
    public toProduct(product:any):Product{
        return new Product(product.pic,product.name,product.category,product.subcategory,product.description,product.price,product.sellerId,product.sellerAdress,product.id)
    }
}

File: Kaighobadi_Neda/Aplicatie/Backend/HomeMadeMarketAPI/HomeMadeMarketAPI/Models/Product.cs
using System;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using Newtonsoft.Json;

namespace HomeMadeMarketAPI.Models
{
    public class Product
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        [JsonProperty("id")]
        public string Id { get; set; }
        [BsonElement("pic")]
        [JsonProperty("pic")]
        public string Pic { get; set; }
        [BsonElement("name")]
        [JsonProperty("name")]
        public string Name {get; set;}
        [BsonElement("category")]
        [JsonProperty("category")]
        public string Category { get; set; }
        [BsonElement("subcategory")]
        [JsonProperty("subcategory")]
        public string Subcategory { get; set; }
        [BsonElement("description")]
        [JsonProperty("description")]
        public string Description { get; set; }
        [BsonElement("price")]
        [JsonProperty("price")]
        public float Price { get; set; }
        [BsonElement("sellerId")]
        [JsonProperty("sellerId")]
        public string SellerId { get; set; }
        [BsonElement("sellerAdress")]
        [JsonProperty("sellerAdress")]
        public string SellerAdress { get; set; }

        //public void changeDescription(string description)
        //{
        //    this.Description = description;
        //}
        //public void changePrice(float price)
        //{
        //    this.Price = price;
        //}
        //public void changeType(string type)
        //{
        //    this.Type = type;
        //}
    }

}
